darshan-runtime/lib/darshan-hdf5.c

```diff
@@ -433,6 +445,15 @@ herr_t DARSHAN_DECL(H5Fclose)(hid_t file_id)
                 tm1, tm2, rec_ref->last_meta_end);
             darshan_delete_record_ref(&(hdf5_file_runtime->hid_hash),
                 &file_id, sizeof(hid_t));
+
+#ifdef HAVE_LDMS
+    rec_ref->close_counts++;
+    /* publish close information for h5f */
+    extern struct darshanConnector dC;
+    if(dC.ldms_lib)
+        if(dC.hdf5_enable_ldms)
+            darshan_ldms_connector_send(rec_ref->file_rec->base_rec.id, rec_ref->file_rec->base_rec.rank, rec_ref->close_counts, "close", -1, -1, -1, -1, rec_ref->file_rec->counters[H5F_FLUSHES],rec_ref->file_rec->fcounters[H5F_F_CLOSE_START_TIMESTAMP], rec_ref->file_rec->fcounters[H5F_F_CLOSE_END_TIMESTAMP], rec_ref->file_rec->fcounters[H5F_F_META_TIME], "H5F", "MOD");
```

Comment:
Okay, 1 more thing that I think is an error that I hadn't caught previously. I don't mind fixing if you can confirm.

You are calling `darshan_ldms_connector_send()` differently here than in other modules. In other modules, you always pass in `tm1` and `tm2` (which are timers for the start and end time of this specific operation), but here you are passing in the `CLOSE_START_TIMESTAMP` and `CLOSE_END_TIMESTAMP` (which measure the time of the very first close and the very last close) instead.

I'm assuming we want to modify the use of this in the HDF5 modules to always use `tm1` and `tm2` instead? Affected functions are H5Fclose, H5Dclose, H5Dwrite, and H5Dread -- H5Fopen and H5Dopen both already are using `tm1` and `tm2`.

Reply:
Yes this is correct. It's supposed to be ``tm1`` and ``tm2``. I actually have some changes that I need to update here (just renaming env variables) so I can fix this and add it to the commit. Thanks for catching this!